src/Esfa.Vacancy.Infrastructure/Services/TrainingDetailService.cs

```diff
@@ -11,16 +11,22 @@ namespace Esfa.Vacancy.Infrastructure.Services
 {
     public class TrainingDetailService : ITrainingDetailService
     {
+        private readonly IProvideSettings _provideSettings;
         private readonly ILog _logger;
-        private readonly string _dasApiBaseUrl;
+        private string _dasApiBaseUrl;
 
         public TrainingDetailService(IProvideSettings provideSettings, ILog logger)
         {
+            _provideSettings = provideSettings;
             _logger = logger;
             _dasApiBaseUrl =
                 provideSettings.GetSetting(ApplicationSettingKeys.DasApprenticeshipInfoApiBaseUrlKey);
         }
 
+        private string DasApiBaseUrl => _dasApiBaseUrl ??
+                                        (_dasApiBaseUrl = _provideSettings.GetSetting(ApplicationSettingKeys.DasApprenticeshipInfoApiBaseUrlKey));
+
```

Comment:
Consciously avoiding any operation in the constructor that could raise error and fail resolution which could lead to wild goose chase like Scott had to. Hence delayed resolving the setting till we actually need it. 

Local variable is required to avoid resolving setting again if retry occurs.

Reply:
Fair enough